### PR TITLE
Feature/take offer integration tests

### PR DESCRIPTION
**Pull request by blabno**

### CODE REVIEW

File: src/main/java/io/bisq/api/BisqProxy.java [hunk 1]
```diff
@@ -205,30 +203,13 @@ public PaymentAccountList getAccountList() {
...
-     */
-    public Tuple2<Optional<OfferDetail>, Optional<BisqProxyError>> getOfferDetail(String offerId) throws Exception {
-        Optional<OfferDetail> result = Optional.empty();
-        Optional<BisqProxyError> error = Optional.empty();
-
-        Optional<Offer> offer = getOffer(offerId);
-        if (!offer.isPresent()) {
-            error = BisqProxyError.getOptional("OfferId not found");
+    public Offer getOffer(String offerId) {
+        final String safeOfferId = null == offerId ? "" : offerId;
```
Comment (mrosseel) [inline src/main/java/io/bisq/api/BisqProxy.java]: looks a bit weird in the beginning, could you rewrite?

final String safeOfferId = (offerId == null) ? "" : offerId;

File: src/main/java/io/bisq/api/model/OfferDetail.java [hunk 2]
```diff
@@ -28,31 +28,34 @@
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
 public class OfferDetail {
     @JsonProperty
-    String offer_id;
+    public String offer_id;
```
Comment (mrosseel) [inline src/main/java/io/bisq/api/model/OfferDetail.java]: why did you need to put them public?

File: src/test/java/io/bisq/api/ApiTestHelper.java [hunk 3]
```diff
@@ -98,10 +98,16 @@ public static SepaAccountToCreate randomValidCreateSepaAccountPayload() {
         accountToCreate.holderName = faker.name().fullName();
         accountToCreate.countryCode = countryCode;
         accountToCreate.selectedTradeCurrency = faker.options().option("PLN", "USD", "EUR", "GBP");
+        if (null != tradeCurrency)
```
Comment (mrosseel) [inline src/test/java/io/bisq/api/ApiTestHelper.java]: I make my null checks the other way round, suppose neither order is right/wrong

File: src/test/java/io/bisq/api/ApiTestHelper.java [hunk 4]
```diff
@@ -133,4 +139,68 @@ public static void sendFunds(Container bitcoin, String walletAddress, double amo
...
+    public static void deselectAllArbitrators(int apiPort) {
+        getArbitrators(apiPort).stream().forEach(arbitratorAddress -> deselectArbitrator(apiPort, arbitratorAddress));
+    }
+
+    private static ValidatableResponse deselectArbitrator(int apiPort, String arbitratorAddress) {
+        return given().
+                port(apiPort).
+//
+        when().
+                        post("/api/v1/arbitrators/" + arbitratorAddress + "/deselect").
```
Comment (mrosseel) [inline src/test/java/io/bisq/api/ApiTestHelper.java]: just noticed this, maybe it's a good idea to create a final static field somewhere with this path, or at least the /api/v1 path.

File: support/offer_make.sh [hunk 5]
```diff
@@ -1,2 +1,11 @@
-clear
-curl -s "http://localhost:8080/api/v1/offer_make?market=xmr_btc&payment_account_id=$1&direction=BUY&amount=100000000&min_amount=50000000&fixed=fixed&price=600" | python -m json.tool | less
+#!/usr/bin/env bash
```
Comment (mrosseel) [inline support/offer_make.sh]: why did you update this? Do you use it somewhere?

File: src/main/java/io/bisq/api/BisqProxy.java (hunk 1 above)
Comment (blabno) [reply src/main/java/io/bisq/api/BisqProxy.java]: Do you mean just adding parentheses?

Comment (mrosseel) [reply src/main/java/io/bisq/api/BisqProxy.java]: yes that should  be enough, if offerId came first maybe it would be more readable, but parentheses for sure

File: src/test/java/io/bisq/api/ApiTestHelper.java (hunk 3 above)
Comment (blabno) [reply src/test/java/io/bisq/api/ApiTestHelper.java]: I'm using Yoda notation which is a very good habit. Look at this: `variable.equals("constant")`, you ask yourselves for NPE. `"constant".equals(variable)` is much more secure.

File: support/offer_make.sh (hunk 5 above)
Comment (blabno) [reply support/offer_make.sh]: yes, I've adjusted the calls to our docker-compose setup and new endpoint definition.

File: src/test/java/io/bisq/api/ApiTestHelper.java (hunk 3 above)
Comment (mrosseel) [reply src/test/java/io/bisq/api/ApiTestHelper.java]: yes I also do that, but in this case it doesn't make a difference. You could argue that it's more consistent to view null as the constant and do it the same way.

File: src/main/java/io/bisq/api/model/OfferDetail.java (hunk 2 above)
Comment (blabno) [reply src/main/java/io/bisq/api/model/OfferDetail.java]: Must be public so that rest assured can use it to construct object based on JSON from the response.

File: support/offer_make.sh (hunk 5 above)
Comment (mrosseel) [reply support/offer_make.sh]: do you personally use these scripts for something?

Comment (blabno) [reply support/offer_make.sh]: Yes, I use it during test development so that I can run API outside of docker and debug.
Those scripts allow me to seed initial state for different stages of trade process.
